### PR TITLE
Move the creation of Runtime dot graph to a separate submodule

### PR DESCRIPTION
**Pull request by rbehjati**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by rbehjati** [comment]
I though `oak/server/rust/oak_runtime/src/runtime/mod.rs` is getting too big. So I moved the stuff related to the creation of the dot graph to a separate submodule. 
What do you think @daviddrysdale, @tiziano88?

File: oak/server/rust/oak_runtime/src/runtime/graph.rs [hunk 1]
```diff
@@ -0,0 +1,271 @@
+use crate::runtime::{ChannelHalfDirection, NodeId, Runtime};
```
Comment (daviddrysdale) [inline oak/server/rust/oak_runtime/src/runtime/graph.rs:1]: Need top-level license text, and a module comment would be nice.

File: oak/server/rust/oak_runtime/src/runtime/graph.rs [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+impl HtmlPath for (NodeId, oak_abi::Handle) {
+    fn html_path(&self) -> String {
+        format!("{}/{}", self.0.html_path(), self.1)
+    }
+}
+
+pub trait DotGraph {
+    fn graph(&self) -> String;
+
```
Comment (daviddrysdale) [inline oak/server/rust/oak_runtime/src/runtime/graph.rs:43]: My personal preference would be to skip the blank lines in this `trait` definition, but up to you.

File: oak/server/rust/oak_runtime/src/runtime/graph.rs (hunk 1 above)
Comment (rbehjati) [reply oak/server/rust/oak_runtime/src/runtime/graph.rs:19]: Thanks. Added.

File: oak/server/rust/oak_runtime/src/runtime/graph.rs (hunk 2 above)
Comment (rbehjati) [reply oak/server/rust/oak_runtime/src/runtime/graph.rs]: Thanks. Fixed.

**Pull request merged**